### DISCUSSION TITLE
Struggling to get this working with `vue-cli-plugin-electron-builder`

### DISCUSSION THREAD

**Discussion by Jazcash** [posva/unplugin-vue-router#92] [Questions and Help] [answered]
My `vue.config.js` contains:
```
configureWebpack: {
    plugins: [
        require("unplugin-vue-router/webpack")({ routesFolder: "src/views" })
    ],
},
```
But that gives me the following build error:

![image](https://user-images.githubusercontent.com/1434248/201787411-cdf5fb90-20cf-4d56-aee0-cf12c467b998.png)

If I do `require("unplugin-vue-router/webpack").default({ routesFolder: "src/views" })` then it loads in fine at least, but then I get:

![image](https://user-images.githubusercontent.com/1434248/201787562-3edb51cb-a432-47f8-91a3-fb16ae36d327.png)

The HTML file it's referencing is in a different dir to all my `.vue` pages, so I'm confused why this plugin is even trying to process it. Any ideas?

**Comment by posva** [comment]
The error seems about using require. Have you tried using import at the top of the file?

**Reply by Jazcash** [reply]
Yes, that works, and then I get the error at the bottom

**Comment by lermontex** [comment]
@posva, I have exactly the same error when trying to use Vue CLI.


```
// vue.config.js
const routerPlugin = require("unplugin-vue-router/webpack").default;

module.exports = {
  configureWebpack: {
    plugins: [
      routerPlugin({
        routesFolder: "src/pages",
      }),
    ],
  },
};
```
--->

```
 ERROR  Failed to compile with 1 error08:14:48

  Error: Child compilation failed:
  Module parse failed: Unexpected token (1:0)
  File was processed with these loaders:
   * ./node_modules/html-webpack-plugin/lib/loader.js
   * ./node_modules/unplugin/dist/webpack/loaders/load.js
  You may need an additional loader to handle the result of these loaders.
  > <!DOCTYPE html>
  | <html lang="">
  |   <head>
  ModuleParseError: Module parse failed: Unexpected token (1:0)
  File was processed with these loaders:
   * ./node_modules/html-webpack-plugin/lib/loader.js
   * ./node_modules/unplugin/dist/webpack/loaders/load.js
  You may need an additional loader to handle the result of these loaders.
  > <!DOCTYPE html>
```

This file is located in the "public" folder

```
/public/index.html
src/pages/
```

**Comment by soc221b** [comment, marked as answer]
I also encountered this problem. 😞 

@posva I found out that can be fixed by adding the following script to `src/index.ts`, but IDKW returning `undefined` from load will cause this error: 🤔 

```diff
    // TODO: is it worth having a loadInclude?
+   loadInclude(id) {
+     return id.endsWith('.html') === false
+   },
```

**Reply by posva** [reply]
You are right, the `loadInclude()` is necessary for webpack so I added it and released a fix patch
